spec/requests/users_spec.rb

```diff
@@ -51,4 +51,87 @@
       end
     end
   end
+
+  describe '#edit_user_request' do
+    image_path = File.join(File.dirname(__FILE__), '..', 'images', 'download.jpeg')
+    let(:user) { create :user }
+    let(:modified_user) { attributes_for :user }
+
+    it 'with no logged user - redirects to login' do
+      get edit_user_registration_path
+      expect(response).to redirect_to(new_user_session_path)
+    end
+
+    context 'with logged user' do
+      before do
+        user.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+        sign_in user
+        get edit_user_registration_path, params: { user: modified_user }
+      end
+
+      it 'renders edit page' do
+        expect(response).to have_http_status(:success)
+      end
+    end
+  end
+
+  describe '#update_user_request' do
+    image_path = File.join(File.dirname(__FILE__), '..', 'images', 'download.jpeg')
+    let(:user) { create :user, password: 'password' }
+    let(:modified_user) { attributes_for :user }
+    let(:user2) { create :user, password: 'password' }
+
+    before do
+      user.avatar.attach(io: File.open(image_path), filename: 'download.jpeg', content_type: 'image/jpeg')
+      sign_in user
+
+      new_image_path = File.join(File.dirname(__FILE__), '..', 'images', 'default.webp')
+      modified_user[:avatar] = fixture_file_upload(new_image_path, 'image/webp')
+    end
+
+    it 'redirects to user show page - wrong password' do
+      put user_registration_path, params: { user: modified_user.merge(current_password: 'wrongpassword') }
+      expect(response).to render_template('edit')
+    end
+
+    it 'redirects to user show page - used email' do
+      put user_registration_path, params: { user: { email: user2.email } }
+      expect(response).to render_template('edit')
+    end
+
+    it 'redirects to user show page - used nickname' do
```

Comment:
These two tests are passing because you didn't provide the current password! Not because the reason you want them to actually redirect you